src/components/pages/schedule-page/schedule-event-edit-section/schedule-form/components/schedule-dropdown/TimeScheduleDropdown.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { FC, useEffect } from 'react';
+import dayjs from 'dayjs';
+import { useField } from 'formik';
+
+import Dropdown from '@/components/common/ui/form/dropdown';
+import { FormikDropdownProps } from '@/components/common/ui/form/with-formik/dropdown/FormikDropdown';
+
+import { getDateWithTimeSetted } from '../../utils/getDateWithTimeSetted';
+
+import * as styles from './ScheduleDropdown.styles';
+
+interface TimeScheduleDropdownProps extends FormikDropdownProps {
+  date: dayjs.Dayjs;
+}
+
+export const TimeScheduleDropdown: FC<TimeScheduleDropdownProps> = ({
+  clearOnUnmount,
+  name,
+  date,
+  ...props
+}) => {
+  const [{ value }, { touched, error }, { setValue, setTouched }] =
+    useField(name);
+
+  React.useEffect(() => {
+    setValue(getDateWithTimeSetted(date, value));
+  }, [date, value, setValue]);
+
+  const onChange = (option: string) => {
+    setTouched(true);
+    setValue(option);
+  };
+
+  return (
+    <Dropdown
+      {...props}
+      value={value}
+      touched={touched}
+      error={error}
+      showRemark={!!error}
+      dropdownSx={styles.dropdown}
+      inputSx={styles.input()}
```

Comment:
why input is called as function?

Reply:
because it is a function, i guess, and why is that i actually don`t know, because it isn`t written by me
![image](https://github.com/fictadvisor/fictadvisor-web/assets/131526240/ee04a029-f4bf-4e9f-bad8-baf8fee91b1f)